wallet/src/de/schildbach/wallet/service/CoinJoinService.kt

```diff
@@ -239,21 +254,43 @@ class CoinJoinMixingService @Inject constructor(
         this.mode = mode
     }
 
-    override fun configureMixing(
+    override suspend fun configureMixing(
         amount: Coin,
         requestKeyParameter: RequestKeyParameter,
         requestDecryptedKey: RequestDecryptedKey,
+        restoreFromConfig: Boolean
     ) {
-        CoinJoinClientOptions.setSessions(4)
-        CoinJoinClientOptions.setAmount(amount)
-        CoinJoinClientOptions.setMultiSessionEnabled(false)
-
-        when (mode) {
-            CoinJoinMode.BASIC -> CoinJoinClientOptions.setAmount(Coin.ZERO)
-            CoinJoinMode.INTERMEDIATE -> CoinJoinClientOptions.setRounds(1)
-            CoinJoinMode.ADVANCED -> CoinJoinClientOptions.setRounds(2)
-        }
+        if (restoreFromConfig) {
+            // read from data store
+            val amountToMix = config.get(CoinJoinConfig.COINJOIN_AMOUNT)
+            val rounds = config.get(CoinJoinConfig.COINJOIN_ROUNDS)
+            val sessions = config.get(CoinJoinConfig.COINJOIN_SESSIONS)
+            val isMultiSession = config.get(CoinJoinConfig.COINJOIN_MULTISESSION)
```

Comment:
If the datastore config has a bunch of settings that we often get/observe and save all at once, it might be more convenient to create a data class that encapsulates them and separate methods for read/write instead of generic `get` and `set`.

See `ExploreConfig` for the example.

Reply:
Yes, this will be great.